src/verilog/verilog_preprocessor_tokenizer.cpp

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************\
+
+Module: Verilog Preprocessor Tokenizer
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+#include "verilog_preprocessor_tokenizer.h"
+
+#include <util/invariant.h>
+
+void verilog_preprocessor_token_sourcet::skip_until_eol()
+{
+  while(true)
+  {
+    auto token = peek();
+    if(token == '\n' || token.is_eof())
+      break;
+    next_token(); // eat
+  }
+}
+
+void verilog_preprocessor_token_sourcet::skip_ws()
+{
+  while(true)
+  {
+    auto token = peek();
+    if(token != ' ' && token != '\t')
+      break;
+    next_token(); // eat
+  }
+}
+
+std::string
+verilog_preprocessor_token_sourcet::tokent::string_literal_value() const
+{
+  PRECONDITION(kind == STRING_LITERAL);
+  PRECONDITION(text.size() >= 2);
+  std::string result;
+  result.reserve(text.size());
+
+  for(std::size_t i = 1; i < text.size() - 1; i++)
+  {
+    char ch = text[i];
+    result += ch;
+  }
+
+  return result;
```

Comment:
How about `return text.substr(1, text.size() - 2));`

Reply:
Apologies, this is missing a bit. I'll need to add un-escaping for `\t`, `\n`, etc.